pkg/trace/stats/aggregation.go

```diff
@@ -117,14 +121,49 @@ func peerTagsHash(tags []string) uint64 {
 func NewAggregationFromGroup(g *pb.ClientGroupedStats) Aggregation {
 	return Aggregation{
 		BucketsAggregationKey: BucketsAggregationKey{
-			Resource:     g.Resource,
-			Service:      g.Service,
-			Name:         g.Name,
-			SpanKind:     g.SpanKind,
-			StatusCode:   g.HTTPStatusCode,
-			Synthetics:   g.Synthetics,
-			PeerTagsHash: peerTagsHash(g.PeerTags),
-			IsTraceRoot:  g.IsTraceRoot,
+			Resource:       g.Resource,
+			Service:        g.Service,
+			Name:           g.Name,
+			SpanKind:       g.SpanKind,
+			StatusCode:     g.HTTPStatusCode,
+			Synthetics:     g.Synthetics,
+			PeerTagsHash:   peerTagsHash(g.PeerTags),
+			IsTraceRoot:    g.IsTraceRoot,
+			GRPCStatusCode: g.GRPCStatusCode,
 		},
 	}
 }
+
+func getGRPCStatusCode(meta map[string]string, metrics map[string]float64) string {
+	// List of possible keys to check in order
+	metaKeys := []string{"rpc.grpc.status_code", "grpc.code", "rpc.grpc.status.code", "grpc.status.code"}
+
+	for _, key := range metaKeys {
+		if strC, exists := meta[key]; exists && strC != "" {
+			c, err := strconv.ParseUint(strC, 10, 32)
+			if err == nil {
+				return strconv.FormatUint(c, 10)
+			}
+
+			if strings.ToUpper(strC) == "CANCELED" || strings.ToUpper(strC) == "CANCELLED" { // the rpc code google api checks for "CANCELLED" but we receive "Canceled" from upstream
```

Comment:
can u create a local variable for `strings.ToUpper(strC)` it is used in multiple places.